### DISCUSSION TITLE
How does winget configuration deal with state conflicts

### DISCUSSION THREAD

**Discussion by PilotBob** [microsoft/winget-cli#3324] [General] [answered]
Let's say for example I have two dsc configuration files. 

They each set a windows setting to a different value. 

If I try to apply the configuration that conflicts with an existing deployed DSC, does it stop and show me the conflict, or will it happily change the setting breaking the previous DSC configuration?

**Comment by denelon** [comment, marked as answer]
No, the current implementation is simply looking at what you are currently attempting to configure.

We're recording the configurations that have been previously applied, but we aren't taking action on it yet. I have some internal documents on thoughts for what we could/should do with WinGet configuration. I'll start creating more of those issues now.

**Reply by denelon** [reply]
You can run `winget configure test <path to configuration file>` and the test method is invoked on the resource. From there you can see if anything isn't in the desired state.

**Comment by denelon** [comment]
Related Issue:
* https://github.com/microsoft/winget-cli/issues/3325
